packages/flutter_tools/lib/src/windows/windows_version_validator.dart

```diff
@@ -106,8 +119,22 @@ class ProcessLister {
 
   final ProcessManager processManager;
 
+  static const String powershell = 'powershell';
+  static const String pwsh = 'pwsh';
+
+  bool canRunPowershell() {
+    return processManager.canRun(powershell) || processManager.canRun(pwsh);
+  }
+
   Future<ProcessResult> getProcessesWithPath() async {
     const String argument = 'Get-Process | Format-List Path';
-    return processManager.run(<String>['powershell', '-command', argument]);
+    const List<String> psArgs = <String>['-command', argument];
+    if (processManager.canRun(powershell)) {
+      return processManager.run(<String>[powershell, ...psArgs]);
+    }
+    if (processManager.canRun(pwsh)) {
+      return processManager.run(<String>[pwsh, ...psArgs]);
+    }
+    throw StateError('Failed to find $powershell or $pwsh on PATH!');
```

Comment:
Unless `!` is some notation I am not aware of, we shouldn't yell at the user (per the style guide, IIRC)

```suggestion
    throw StateError('Failed to find $powershell or $pwsh on PATH');
```